### ISSUE TITLE
Allow quick pick items to always be visible and sticky

### ISSUE THREAD

**Issue by misolori** [microsoft/vscode#131128]
Based on internal discussions, would like to have an option in the Quick Pick item API to allow certain items to be grouped together and separated by a border. Additionally, when there are long entries in the list, it would be great to have this group always visible like below (see items at the bottom of the quick pick):

<img width="648" alt="CleanShot 2021-08-18 at 13 53 40@2x" src="https://user-images.githubusercontent.com/35271042/129970755-5b43f2ea-2f2c-4076-a6ef-5df7f3cd0afd.png">

And when filtering the list, it would filter as it normally does:

<img width="633" alt="CleanShot 2021-08-18 at 13 56 13@2x" src="https://user-images.githubusercontent.com/35271042/129971045-3907bf17-db01-458c-a1b1-999c3e76f2a7.png">

<img width="627" alt="CleanShot 2021-08-18 at 13 56 20@2x" src="https://user-images.githubusercontent.com/35271042/129971057-c2318503-8641-468c-98c1-8ba39928faf2.png">

cc @luabud

**Comment by TylerLeonhardt** [comment]
> And when filtering the list, it would filter as it normally does:

It's interesting, I expected the opposite from the sticky items. There would only be one or two of them so they really are just "extra actions I always want to see" while the non-sticky bit is the "true list I'm filtering through"

**Comment by misolori** [comment]
I can see that. How would you navigate to those actions via keyboard though if the list is long? You'd have to go through each item in the list or go through the tab stops, which feels like a bit of a hack for something the quick pick already supports. I just imagined it would be faster to type a few characters + enter.

**Comment by TylerLeonhardt** [comment]
Yeah I think that makes sense. Right now we have a concept of separators in Core... I think supporting separators in the API is a prereq of this work because ideally we can use separators to determine where the sticky starts (aka the last separator would denote stickiness).

**Comment by misolori** [comment]
This might be solved via https://github.com/microsoft/vscode/issues/149057 with having section headers and actions

**Comment by karrtikr** [comment]
I can help test this, is there an option that we can specify within the `QuickPickItem` object?

**Comment by TylerLeonhardt** [comment]
This was closed on accident
